### ISSUE TITLE
[meson] feature request

### ISSUE THREAD

**Issue by udeved** [OpenRC/openrc#442]
Could the meson system have an option to set the rc_libexecdir?
We have set it /usr/lib/openrc, meson has it hardcoded to 'rc'.
A made a patch for that, I can do a PR, I was trying to fix the sysvinit issue, which is now fixed already in 0.44.1  :+1:  

Another issue, I haven't spotted any pam install code in meson, with 0.44, /etc/pam.d/ wasn't installed, no pam files for ssd and supervisor despite pam set true.

**Comment by williamh** [comment]
The pam issue definitely needs to be fixed; I'll take a look at that.

You should be able to use the --libexecdir option to meson to change the location of libexecdir, but it will add the last component as "rc"; I kept that from the previous build system. Were you overriding it before and changing it to /usr/lib/openrc?

To be honest I've thought about changing the default from "rc" to "openrc" in this path, but I'm cautious because I don't want to break anything.

**Comment by udeved** [comment]
Yes, I set /usr/lib/openrc on artix by default.
If I was to switch to meson, the update would probably screw up the running system of users.

https://gitea.artixlinux.org/packagesO/openrc/src/branch/master/trunk/PKGBUILD

**Comment by udeved** [comment]
Great, pam is fixed with latest commit, I added this patch

```
 meson.build       | 2 +-
 meson_options.txt | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 3b787086..c5288c1b 100644
--- a/meson.build
+++ b/meson.build
@@ -85,7 +85,7 @@ else
 endif
 libexecdir = split_usr ? rootprefix / libexecpath :
   get_option('prefix') / libexecpath
-rc_libexecdir = libexecdir / 'rc'
+rc_libexecdir = libexecdir / get_option('librcdir')
 sbindir = rootprefix / get_option('sbindir')

 selinux_dep = dependency('libselinux', required : get_option('selinux'))
diff --git a/meson_options.txt b/meson_options.txt
index e4e7c4df..25d873d2 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -35,3 +35,5 @@ option('termcap', type : 'combo',
   description : 'the termcap library to use')
 option('zsh-completions', type : 'boolean',
   description : 'install zsh completions')
+option('librcdir', type : 'string', value : 'rc',
+  description : 'default location of rc libexec dir')

```

and I get no diff in the actual package built to previous package. :)

**Comment by williamh** [comment]
I'm going to release 0.44.2 to fix the pam issue.

I started looking into the rc_libexecdir changes, but it is a bit more tricky. for non-linux systems, rc_svcdir is tied to rc_libexecdir, so if that is changed it will break them. Feel free to open a pr with your change and I'll look at it.

**Comment by udeved** [comment]
Ok, I send a PR, but the patch just sets the actual name, it still defaults to 'rc', I set it in PKGBUILD to 'openrc' and proper path for rc_libexecdir is made, ie /usr/lib/openrc

Dunno, but if rc_svcdir uses rc_libexecdir, all should be fine, since path is by default libexecdir/rc

https://github.com/OpenRC/openrc/pull/443

**Comment by udeved** [comment]
The build log with 044.2 points out invalid symlinks if sysvinit is used, it has absolute path to the build location.

```
Checking openrc-0.44.2-1-x86_64.pkg.tar.zst
openrc W: Directory (usr/lib/openrc/tmp) is empty
openrc W: Directory (usr/lib/binfmt.d) is empty
openrc W: Directory (usr/lib/openrc/cache) is empty
openrc E: BSD2 is not a common license (it's not in /usr/share/licenses/common/)
openrc E: Symlink (usr/bin/halt) points to non-existing /build/openrc/pkg/openrc/usr/lib/openrc/bin/halt
openrc E: Symlink (usr/bin/poweroff) points to non-existing /build/openrc/pkg/openrc/usr/lib/openrc/bin/poweroff
openrc E: Symlink (usr/bin/rc-sstat) points to non-existing /build/openrc/pkg/openrc/usr/lib/openrc/bin/rc-sstat
openrc E: Symlink (usr/bin/reboot) points to non-existing /build/openrc/pkg/openrc/usr/lib/openrc/bin/reboot
openrc E: Symlink (usr/bin/shutdown) points to non-existing /build/openrc/pkg/openrc/usr/lib/openrc/bin/shutdown
```

**Comment by udeved** [comment]
https://github.com/OpenRC/openrc/pull/444 fixes meson script symlnks

**Comment by williamh** [comment]
I looked at #443, and I can't really accept it in that form.
The problem will come on a non-linux system, because, in that environment, `RC_SVCDIR` points to `RC_LIBEXECDIR/init.d`, and this is where OpenRC keeps track of the state of the system (service dependencies, which services are started/stopped, etc).
If a non-linux user uses that override to change the librc directory, they will also change the location of `RC_SVCDIR`; this would leave OpenRC not knowing the status of any services.
This is basically blocked until  I figure out how to resolve #339; I can probably do this once that is done.

Also, thinking about it, I am curious why you can't move to the default libexec directory since everything in that directory is considered private to openrc service scripts.

On Linux, the /run/openrc directory is where state information goes, so you should be able to update with no issues.

Can you test and let me know? I'm really curious why this directory changing names is a blocker for you.

**Comment by udeved** [comment]
Our /usr/lib/openrc distro default is much caused by the fact we also have runit and s6 in the repos, which also have a rc component. This is to avoid any confusion for users, it makes it clear from the look at it, it is openrc related.
